src/hooks/useStakingSlider.ts

```diff
@@ -0,0 +1,53 @@
+import { useLocation } from 'react-router-dom';
+import { useGetColonyActionQuery } from '~gql';
+
+const getTransactionHashFromPathName = (pathname: string) =>
+  pathname.split('/').pop();
+
+const useStakingSlider = (isObjection: boolean) => {
+  const { pathname } = useLocation();
+  const transactionHash = getTransactionHashFromPathName(pathname);
+  const { data: actionData } = useGetColonyActionQuery({
```

Comment:
Rerunning the colony action query inside the staking slider just gets the data from the cache (`cache-first` is the default network policy), and means we don't have to pass all these props via the objection modal.

Reply:
Good idea to circumvent the context not being available!